[Nguyen Ba Van Nhi] iP

nbvannhi opened the pull request:
# Bloom

> "It is better to light a candle than to curse the darkness." – Anonymous

Bloom frees your mind of having to remember things you need to do. It's,

 * text-based
 * easy to learn
 * ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/nbvannhi/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

---

If you are Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

nbvannhi committed:
Rename bot to Bloom

nbvannhi committed:
Add Greet, Echo, Exit

nbvannhi committed:
Add Add, List

nbvannhi committed:
Add Mark as Done

- Add done() method in Command.java
- Create start() method to start Bloom
- Move all task-related commands to Command.java

nbvannhi committed:
Add toString() method in Task.java

nbvannhi committed:
Add ToDo, Event, Deadline

nbvannhi committed:
Add Delete

nbvannhi committed:
Add Handle Errors

- WrongCommandBloomException is caught when a potential mistaken command is inputted
- IncompleteCommandBloomException is caught when a correct command tag is missing attributes
- InvalidCommandBloomException is caught when an unrecognised command is inputted
- BloomException is the parent class for all Bloom-related exceptions

nbvannhi committed:
Add Automated Text UI Testing

nbvannhi committed:
Add comments, method descriptions

nbvannhi committed:
Add Use Enums

nbvannhi committed:
Fix minor bug

- get() method in Command.java was missing two cases
- Test cases were re-ran and gave expected outputs

nbvannhi committed:
Restructure project files

nbvannhi committed:
Add Save

An java.io.UnsupportedEncodingException is thrown
in saveToDb() in Storage.java has yet to be resolved.

nbvannhi committed:
Add Dates and Times

nbvannhi committed:
Merge branch 'branch-level-7'

nbvannhi committed:
Add More OOP

nbvannhi committed:
Add JUnit Tests for src/main/java/bloom/task/

nbvannhi committed:
Create a JAR File

nbvannhi committed:
Fix indentation of MarkCommand

nbvannhi committed:
Add Find

nbvannhi committed:
Add JavaDoc

nbvannhi committed:
Follow the Coding Standard

@Override annotations are added in the Commands

nbvannhi committed:
Merge branch 'branch-A-CodingStandard'

Conflicts resolved in the child Command files and Task file

nbvannhi committed:
Merge branch 'branch-Level-9'

nbvannhi committed:
Add branch-Level-8 branch

nbvannhi committed:
Merge branch 'branch-Level-8'

nbvannhi committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

nbvannhi committed:
Add Use CheckStyle with Gradle

nbvannhi committed:
Merge branch 'branch-A-Gradle' into branch-A-CheckStyle

nbvannhi committed:
Add GUI

Bloom currently only repeats whatever inputted. According changes are coming.

nbvannhi committed:
Merge branch 'branch-Level-10'

nbvannhi committed:
Override run() method in MarkCommand.java

Bloom can now handle marking several tasks as done.

nbvannhi committed:
Merge branch 'branch-A-Varargs'

kevinmingtarja reviewed (commented):
LGTM! Good work overall and all the best with the remaining tasks :)

ErnestCuong commented on src/main/java/Bloom.java:
```diff
@@ -0,0 +1,155 @@
+    }
+
+    @Override
+    public void start(Stage stage) {
```
Perhaps you can consider breaking this method down by using helper methods to make it shorter?

ErnestCuong commented on src/main/java/bloom/constant/Message.java:
```diff
@@ -0,0 +1,42 @@
+ * Contains messages to communicate with users.
+ */
+
+public enum Message {
```
I like how you make use of enum to group all the messages into one place.

ErnestCuong commented on src/main/java/bloom/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+     */
+
+    public String getStatusIcon() {
+        return (this.isDone ? "X" : " ");
```
Perhaps instead of returning "X" or " ", you can consider returning "[X]" and "[ ]".

ErnestCuong commented on src/main/java/bloom/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+     * @return a string representing the task
+     */
+
+    public String toDb() {
```
Perhaps you can make the method name more descriptive like `toDatabaseString`.

ErnestCuong reviewed (commented):
Looks good to merge. I don't see any major coding violations, perhaps you can consider shortening some of the methods. Great work.

kaixin-hc commented on src/main/java/Bloom.java:
```diff
@@ -0,0 +1,155 @@
+     * Creates two dialog boxes, one echoing user input and the other containing
+     * Bloom's reply and then appends them to the dialog container.
+     * Clears the user input after processing.
+     */
```
I've noticed many of your comments have whitespace between the end of the comment and the start of the method. According to the coding standards, `Punctuation behind each parameter description -No blank line between the documentation block and the method/class` -- perhaps you could consider changing this in your next commit!

kaixin-hc commented on src/main/java/Launcher.java:
```diff
@@ -0,0 +1,11 @@
+import javafx.application.Application;
+
+/**
+ * A launcher class to workaround classpath issues.
```
nit: work around (workaround is a noun)

kaixin-hc reviewed (commented):
Aside from these two nits, LGTM! I thought it was interesting that you used enums to separate out the messages for each command -- but also had individual classes for each command. Since each enum is only used in one place, I'm curious about why you didn't just include it as a string within each command. All the best!

nbvannhi committed:
Fix coding style

Empty lines between method description and method declaration are removed.

nbvannhi committed:
Finish Level-10

Convert return type of run() method in all commands to match with the new JavaFX version.

nbvannhi committed:
Add assertions in parseDate()

nbvannhi committed:
Fix arrow head borderline case of DialogBox.java

nbvannhi committed:
Improve code quality of method start() in Bloom.java

start() method is lengthy.

A lengthy method makes it hard to identify and fix bugs, and
also shows limited level of abstraction.

Let's,
* put formatting of each component in separate methods
* add a script to migrate stress tests to the new format

nbvannhi committed:
Merge pull request #1 from nbvannhi/branch-A-Assertions

Add assertions

nbvannhi committed:
Merge branch 'master' of https://github.com/nbvannhi/ip

* 'master' of https://github.com/nbvannhi/ip:
  Add assertions in parseDate()

nbvannhi committed:
Merge pull request #2 from nbvannhi/branch-A-CodeQuality

Improve code quality

nbvannhi committed:
Merge branch 'master' of https://github.com/nbvannhi/ip

* 'master' of https://github.com/nbvannhi/ip:
  Improve code quality of method start() in Bloom.java
  Fix arrow head borderline case of DialogBox.java

nbvannhi committed:
Add D-Notes to support managing notes attached with each task

nbvannhi committed:
Set up CI

nbvannhi committed:
Use Streams by adding method printUpcomingTaskList() in TaskList.java

nbvannhi committed:
Complete date feature

nbvannhi committed:
Update print date format

nbvannhi committed:
Add A-UserGuide

nbvannhi committed:
Update JUnit test with new date format

nbvannhi committed:
Fix bugs

nbvannhi committed:
Fix bugs

nbvannhi committed:
Remove long method

nbvannhi committed:
Shorten long method

nbvannhi committed:
Fix messages
